index.bs

```diff
@@ -6554,6 +6577,123 @@ requests will be affected.
+The <dfn export>WebDriver BiDi cache behavior</dfn> steps given |request| are:
+
+1. Let |context| be null.
+
+1. If |request|'s [=request/window=] is an [=environment settings object=]:
+
+  1. Let |environment settings| be |request|'s [=request/window=]
+
+  1. If there is a [=/browsing context=] whose [=active window=] is |environment
+     settings|' [=environment settings object/global object=], set |context| to
+     the [=top-level browsing context=] for that browsing context.
+
+1. If |context| is not null and [=navigable cache behavior map=] [=set/contains=]
+   |context|, return [=navigable cache behavior map=][|context|].
+
+1. Return [=default cache behavior=].
+
+</div>
+
+<div algorithm="remote end steps for network.setCacheBehavior">
+The [=remote end steps=] given <var ignore>session</var> and |command parameters| are:
+
+1. Let |behavior| be |command parameters|["<code>cacheBehavior</code>"].
+
+1. If |command parameters| does not [=map/contain=] "<code>contexts</code>":
+
+  1. Set the [=default cache behavior=] to |behavior|.
+
+  1. [=map/Clear=] [=navigable cache behavior map=].
+
+  1. Switch on the value of behavior:
+    <dl>
+      <dt>"<code>bypass</code>"
+      <dd>Perform implementation-defined steps to disable any
+        implementation-specific resource caches for all future
+        network requests.
+      <dt>"<code>default</code>"
+      <dd>Perform implementation-defined steps to enable any
+        implementation-specific resource caches that are usually enabled in the
+        current [=remote end=] configuration for all future network requests.
+    </dl>
+
+  1. Return [=success=] with data null.
+
+1. Let |contexts| be an empty [=/set=].
+
+1. For each |context id| of |command parameters|["<code>contexts</code>"]:
+
+  1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+    with |context id|.
+
+  1. If |context| is not a [=top-level browsing context=], return [=error=]
+     with [=error code=] [=invalid argument=].
+
+  1. [=list/Append=] |context| to |contexts|.
+
+1. For each |context| in |contexts|:
+
+    1. If [=navigable cache behavior map=] [=map/contains=] |context|, and
+       [=navigable cache behavior map=][|context|] is equal to |behavior| then
+       continue.
+
+    1. Switch on the value of behavior:
+      <dl>
+        <dt>"<code>bypass</code>"
+        <dd>Perform implementation-defined steps to disable any implementation-specific
+          resource caches for network requests originating from any browsing
+          context for which |context| is the [=top-level browsing context=].
+        <dt>"<code>default</code>"
+        <dd>Perform implementation-defined steps to enable any
+          implementation-specific resource caches that are usually enabled in the
+          current [=remote end=] configuration for network requests
+          originating from any browsing context for which |context| is the
+          [=top-level browsing context=].
```

Comment:
It's not clear to me if this takes precedence over the global setting or not?

For instance if I first setCacheBehavior "bypass" globally, and then set it to "default" for one context, what should happen?

I believe that the intent is that it should override whatever is the default setting, but since the global usage of the command is supposed to enable/disable implementation specific resource caches "for all future requests", it's hard to tell which one will prevail?

The actual mechanism for enabling/disabling resource caches is not clearly specified so I imagine it's difficult to formalize, but maybe we need a note here to clarify what happens in such a case?

Reply:
I've tried to make this explicit by actually adding a step when we create a context we explicitly disable the caches if it's has a top level context where the caches are explicitly disabled, or if the default is to disable the caches. That assumes that in the absence of that step we'd get the default behavior. I hope that solves the problem well enough for now.

Reply:
That works for me, thanks!